graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +57,214 @@ def get_directives_for_ast(ast):
     return result
 
 
+def get_all_tag_names(ast):
+    """Return a set of strings containing tag names that appear in the query."""
+    return {
+        # Schema validation has ensured this exists
+        directive.arguments[0].value.value
+        for ast, directive in _yield_ast_nodes_with_directives(ast)
+        if directive.name.value == TagDirective.name
+    }
+
+
+def _remove_colocated_tags(non_macro_names, ast):
```

Comment:
This function is pretty long - can we break it down into more modular components that we could then also more easily unit test?